app/namespaces/data_provider/data_provider_controller.py

```diff
@@ -5,43 +5,61 @@
-    def get(self):
-        # Parse pagination request args if they are present
-        sorting_key = request.args.get('sorting_key', None, type=str)
-        reverse = request.args.get('reverse', None, type=bool)
-        page_index = request.args.get('page_index', None, type=int)
-        per_page = request.args.get('per_page', None, type=int)
 
-        # Type must be string not bool, because bool evaluates to true for any non None value including False and True
-        research_fields = False if str.lower(request.args.get('research_fields', 'false', type=str)) == 'false' else True
-        prioritize_estimates = True if str.lower(request.args.get('prioritize_estimates', 'true', type=str)) == 'true' else False
+    def post(self):
+        # Convert input payload to json and throw error if it doesn't exist
+        data = request.get_json()
+        if not data:
+            return {"message": "No input payload provided"}, 400
 
         # Log request info
-        logging.info("Endpoint Type: {type}, Endpoint Path: {path}, Arguments: {args}".format(
+        logging.info("Endpoint Type: {type}, Endpoint Path: {path}, Arguments: {args}, Payload: {payload}".format(
             type=request.environ['REQUEST_METHOD'],
             path=request.environ['PATH_INFO'],
-            args=dict(request.args)))
+            args=dict(request.args),
+            payload=data))
+
+        # All of these params can be empty, in which case, our utility functions will just return all records
+        filters = data.get('filters')
 
-        result = get_filtered_records(research_fields, filters=None, columns=None, start_date=None, end_date=None,
+        # Validate input payload
+        payload, status_code = validate_request_input_against_schema(data, RecordsSchema())
+        if status_code != 200:
+            # If there was an error with the input payload, return the error and 422 response
+            return make_response(payload, status_code)
+
+        sorting_key = data.get('sorting_key')
+        page_index = data.get('page_index')
+        per_page = data.get('per_page')
+        reverse = data.get('reverse')
+        columns = data.get('columns')
+        research_fields = data.get('research_fields')
+        prioritize_estimates = data.get('prioritize_estimates', True)
+        start_date, end_date = convert_start_end_dates(data)
+
+        result = get_filtered_records(research_fields, filters, columns, start_date=start_date, end_date=end_date,
                                       prioritize_estimates=prioritize_estimates)
-        result = jitter_pins(result)
+        if not columns or ("pin_latitude" in columns and "pin_longitude" in columns):
+            result = jitter_pins(result)
 
         # Only paginate if all the pagination parameters have been specified
         if page_index is not None and per_page is not None and sorting_key is not None and reverse is not None:
```

Comment:
tbh I think we can get rid of this block and the `get_paginated_records_list` function. nobody uses these pagination arguments rn and i think it would be clearer if we have one clearly defined scheme for pagination instead of two

Reply:
Great point, `/records` should have no pagination functionality - that should only be in `/records/paginated`. Updated!